pkg/cmd/cli/cmd/env.go

```diff
@@ -371,7 +384,11 @@ func updateEnv(existing []kapi.EnvVar, env []kapi.EnvVar, remove []string) []kap
 		covered.Insert(e.Name)
 		out = append(out, e)
 	}
-	return out
+	envUpdated := true
```

Comment:
I still would rather set it explicitly... I think some of the complexity would be reduced by handling each case (removed, modified, added) in its own loop